numba/tests/test_stencils.py

```diff
@@ -365,12 +407,1272 @@ def test_impl(n):
+        c = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, c, options={'standard_indexing': 'b'})
+
+    # this should fail parfors with a more informative error message
+    # @stencil and @njit do not raise but they should
+    def test_basic53(self):
+        """2 args, standard_indexing on variable that does not exist"""
+        def kernel(a, b):
+            return a[0, 1] + b[0, 2]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        ex = self.exception_dict(pyStencil=ValueError, stencil=Exception, parfor=KeyError, njit=Exception)
+        self.check(kernel, a, b, options={'standard_indexing': 'c'}, expected_exception=ex)
+
+    def test_basic54(self):
+        """2 args, standard_indexing, index from var"""
+        def kernel(a, b):
+            t = 2
+            return a[0, 1] + b[0, t]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+        
+    def test_basic55(self):
+        """2 args, standard_indexing, index from more complex var"""
+        def kernel(a, b):
+            s = 1
+            t = 2 - s
+            return a[0, 1] + b[0, t]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+        
+    def test_basic56(self):
+        """2 args, standard_indexing, added complexity """
+        def kernel(a, b):
+            s = 1
+            acc = 0 
+            for k in b[0, :]:
+                acc += k
+            t = 2 - s - 1
+            return a[0, 1] + b[0, t] + acc
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+        
+    def test_basic57(self):
+        """2 args, standard_indexing, split index operation """
+        def kernel(a, b):
+            c = b[0]
+            return a[0, 1] + c[1]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+
+    # parfors fails
+    def test_basic58(self):
+        """2 args, standard_indexing, split index with broadcast mutation """
+        def kernel(a, b):
+            c = b[0] + 1
+            return a[0, 1] + c[1]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+
+    def test_basic59(self):
+        """3 args, mix of array, relative and standard indexing and const"""
+        def kernel(a, b, c):
+            return a[0, 1] + b[1, 1] + c
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        c = 10
+        self.check(kernel, a, b, c, options={'standard_indexing': ['b', 'c']})
+
+    # parfors fails
+    def test_basic60(self):
```

Comment:
parfors fails to work on this

Reply:
Shouldn't we detect tuples and always have them be standard indexed?

Reply:
Actually, since I don't believe that tuple is really supported by gufunc, I don't think we should support this case.  If you want something like this, then convert to a numpy array, pass that array to the stencil as standard indexed and then access the array or convert to tuple and access it that way.

Reply:
We are just going to raise an exception in parallel=True mode if stencil has a tuple parameter.  If you want this restriction lifted then we suggest just fixing tuples in guvectorize.